src/libraries/Native/Unix/CMakeLists.txt

```diff
@@ -125,9 +135,8 @@ endif ()
 
 if (CMAKE_SYSTEM_NAME STREQUAL Linux)
    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -D_GNU_SOURCE")
-   set(CMAKE_ASM_FLAGS "${CMAKE_ASM_FLAGS} -Wa,--noexecstack")
-   set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--build-id=sha1")
-   set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,--build-id=sha1")
+   add_compile_options($<$<COMPILE_LANGUAGE:ASM>:-Wa,--noexecstack>)
```

Comment:
Same for this, might want to enable it on other ELF targets too.  (Is there a Mach-O equivalent?)

Reply:
Clang on Darwin accepts the --noexecstack too, but looking at llvm source, I can see it is dummy on macOS and doesn't do anything at all. It translates to a call to InitSections on the target object streamer with the bool NoExecStack argument. The macho object streamer ignores this parameter. So I believe there is no such option for macOS.